src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.Android.cs

```diff
@@ -0,0 +1,431 @@
+                if (!File.Exists(path))
+                {
+                    return false;
+                }
+                try {
+                    using (FileStream fs = File.OpenRead(path))
+                    {
+                        LoadTzFile(fs);
+                    }
+                    return true;
+                }
+                catch (IOException) {}
+
+                return false;
+            }
+
+            private unsafe void LoadTzFile(Stream fs)
+            {
+                int headerSize = 24;
+                Span<byte> buffer = stackalloc byte[headerSize];
+                int bytesRead = 0;
+                int bytesLeft = headerSize;
+
+                while (bytesLeft > 0)
+                {
+                    int b = fs.Read(buffer);
+                    if (b == 0)
+                    {
+                        break;
+                    }
+
+                    bytesRead += b;
+                    bytesLeft -= b;
+                }
+
+                AndroidTzDataHeader header = LoadHeader(buffer);
+                ReadIndex(fs, header.indexOffset, header.dataOffset);
+            }
+
+            private unsafe AndroidTzDataHeader LoadHeader(Span<byte> buffer)
+            {
+                // tzdata files are expected to start with the form of "tzdata2012f\0" depending on the year of the tzdata used which is 2012 in this example
+                // since we're not differntiating on year, check for tzdata and the ending \0
+                Span<byte> signature = buffer.Slice(0, 12);
+                Span<byte> tzDataSignature = stackalloc byte[8];
+                tzDataSignature.Clear();
+
+                signature.Slice(0, 6).CopyTo(tzDataSignature);
+                ulong tzdata = (ulong)TZif_ToInt64(tzDataSignature);
+
+                char[] cbits = Encoding.Default.GetChars(tzDataSignature.ToArray());
+
+                string sigBits = "";
+                for (int i = 0; i < signature.Length; i++)
+                {
+                    sigBits += signature[i].ToString() + " ";
+                }
+
+                if (tzdata != 0x747A646174610000 || signature[11] != 0)
+                {
+                    // 0x747A646174640000 = {0x74, 0x7A, 0x64, 0x61, 0x74, 0x64, 0x00, 0x00} = "tzdata[NUL][NUL]"
+                    var b = new StringBuilder(signature.Length);
+                    for (int i = 0; i < signature.Length; ++i)
+                    {
+                        b.Append(' ').Append(HexConverter.ToCharLower(signature[i]));
+                    }
+
+                    throw new InvalidOperationException(SR.Format(SR.InvalidOperation_BadTZHeader, TimeZoneFileName, b.ToString()));
+                }
+
+                AndroidTzDataHeader header;
+
+                header.indexOffset = TZif_ToInt32(buffer.Slice(12, 4));
+                header.dataOffset = TZif_ToInt32(buffer.Slice(16, 4));
+                header.finalOffset = TZif_ToInt32(buffer.Slice(20, 4));
```

Comment:
Is `finalOffset` used? Can we just delete it, since it doesn't appear to be? Then the struct only needs to be 2 ints.

Reply:
Removed, left as a comment to help explain why the header size is 24 bytes